pkg/interactor/save_data_test.go

```diff
@@ -0,0 +1,81 @@
+package interactor
+
+import (
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/CESARBR/knot-cloud-storage/pkg/entities"
+	"github.com/CESARBR/knot-cloud-storage/pkg/mocks"
+	"github.com/stretchr/testify/assert"
+)
+
+var date = time.Date(2020, time.April, 15, 0, 0, 0, 0, time.UTC)
+
+type SaveDataTestCase struct {
+	name             string
+	authorization    string
+	deviceID         string
+	payload          []entities.Payload
+	data             entities.Data
+	fakeLogger       *mocks.FakeLogger
+	fakeDataStore    *mocks.FakeDataStore
+	fakeThingService *mocks.FakeThingService
+}
+
+var sdCases = []SaveDataTestCase{
+	{
+		"authorization key not provided",
```

Comment:
@esdrasjnr has worked very hard on **babeltower's** tests... he knows better than I do, but I believe the best test examples can be found in the [`auth_thing_test.go`](https://github.com/CESARBR/knot-babeltower/blob/master/pkg/thing/interactors/auth_thing_test.go). Please correct me if I'm wrong, Esdras! :eyes:

Reply:
Also, I noticed that you use `authorization key` when referring to the `authorization` parameter. I strongly recommend that its updated to `authorization token` instead. See [this issue](https://github.com/CESARBR/knot-babeltower/issues/68#issue-609435403).

Reply:
I think it could be added to this PR.

Reply:
@Igoraamc I saw you added the commit _3efca15cd7fa4e8dd2f945ac5522e2516222b933_, thanks!
The only thing I would change is to remove the commit message body since the title is already enough.

Reply:
Done!